lib/src/component_declaration/component_base.dart

```diff
@@ -444,6 +444,18 @@ abstract class UiStatefulComponent<TProps extends UiProps, TState extends UiStat
   // ----------------------------------------------------------------------
 }
 
+class WarnOnModify<K, V> extends MapView<K, V> {
+  WarnOnModify(Map componentData): super(componentData);
+
+  @override
+  operator []=(K key, V value) {
+    assert(
```

Comment:
This shouldn't be an assert, since we don't want this to throw and break existing consumer code during development. 

```dart
operator []=(K key, V value) {
  window.console.warn(...)
  super[key] = value;
}
```